modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqexecEngine.scala

```diff
@@ -74,81 +71,74 @@ class SeqexecEngine(settings: SeqexecEngine.Settings) {
+
+  def setWaterVapor(q: EventQueue, wv: WaterVapor, user: UserDetails): IO[Either[SeqexecFailure, Unit]] =
+    q.enqueue1(Event.logDebugMsg("SeqexecEngine: Setting water vapor")) *>
+    q.enqueue1(Event.modifyState[executeEngine.ConcreteTypes]((Engine.State.userDataL ^|-> EngineMetadata.conditionsL ^|-> Conditions.wv).set(wv), SetWaterVapor(wv, user.some))).map(_.asRight)
+
+  def setSkyBackground(q: EventQueue, sb: SkyBackground, user: UserDetails): IO[Either[SeqexecFailure, Unit]] =
+    q.enqueue1(Event.logDebugMsg("SeqexecEngine: Setting sky background")) *>
+    q.enqueue1(Event.modifyState[executeEngine.ConcreteTypes]((Engine.State.userDataL ^|-> EngineMetadata.conditionsL ^|-> Conditions.sb).set(sb), SetSkyBackground(sb, user.some))).map(_.asRight)
+
+  def setCloudCover(q: EventQueue, cc: CloudCover, user: UserDetails): IO[Either[SeqexecFailure, Unit]] =
+    q.enqueue1(Event.logDebugMsg("SeqexecEngine: Setting cloud cover")) *>
+    q.enqueue1(Event.modifyState[executeEngine.ConcreteTypes]((Engine.State.userDataL ^|-> EngineMetadata.conditionsL ^|-> Conditions.cc).set(cc), SetCloudCover(cc, user.some))).map(_.asRight)
 
   def setSkipMark(q: EventQueue,
                   seqId: SPObservationID,
                   user: UserDetails,
                   stepId: edu.gemini.seqexec.engine.Step.Id,
-                  v: Boolean): Task[SeqexecFailure \/ Unit] =
-  q.enqueueOne(Event.skip(seqId.stringValue(), user, stepId, v)).map(_.right)
+                  v: Boolean): IO[Either[SeqexecFailure, Unit]] =
+    q.enqueue1(Event.skip(seqId.stringValue(), user, stepId, v)).map(_.asRight)
 
-  def requestRefresh(q: EventQueue, clientId: ClientID): Task[Unit] = q.enqueueOne(Event.poll(clientId))
+  def requestRefresh(q: EventQueue, clientId: ClientID): IO[Unit] = q.enqueue1(Event.poll(clientId))
 
-  def seqQueueRefreshProcess: Process[Task, executeEngine.EventType] =
-    awakeEvery(settings.odbQueuePollingInterval)(Strategy.DefaultStrategy, DefaultScheduler).map(_ =>
-      Event.getState[executeEngine.ConcreteTypes](refreshSequenceList()))
+  def seqQueueRefreshStream: Stream[IO, executeEngine.EventType] =
+    Scheduler[IO](corePoolSize = 1).flatMap { scheduler =>
+      val fd = Duration(settings.odbQueuePollingInterval.toSeconds, TimeUnit.SECONDS)
+      scheduler.fixedRate[IO](fd).flatMap { _ =>
+        Stream.emit(Event.getState[executeEngine.ConcreteTypes](refreshSequenceList()))
+      }
+    }
 
-  def eventProcess(q: EventQueue): Process[Task, SeqexecEvent] =
-    executeEngine.process(wye(q.dequeue, seqQueueRefreshProcess)(mergeHaltBoth))(Engine.State.empty[EngineMetadata](EngineMetadata.default)).flatMap(x => Process.eval(notifyODB(x))).map {
+  def eventStream(q: EventQueue): Stream[IO, SeqexecEvent] = {
+    executeEngine.process(q.dequeue.merge(seqQueueRefreshStream))(Engine.State.empty[EngineMetadata](EngineMetadata.default)).flatMap(x => Stream.eval(notifyODB(x))).map {
```

Comment:
`mergeHaltBoth` was used to be able to stop the Stream by closing `q`. That is not possible if `merge` is used, because `seqQueueRefreshStream` is an infinite Stream and the Stream produced by `merge` only stops when both `Stream`s stop.

Reply:
Feel free to change it